concent_api/core/tests/test_core_views.py

```diff
@@ -286,6 +286,70 @@ def test_send_reject_message_save_as_receive_out_of_band_status(self):
         self.assertEqual(reject_response.status_code, 202)
         self.assertEqual(Message.objects.last().type, message.RejectReportComputedTask.TYPE)
 
+    @freeze_time("2017-11-17 10:00:00")
+    def test_send_should_return_http_400_if_task_to_compute_deadline_is_negative_integer(self):
+        compute_task_def = message.ComputeTaskDef()
+        compute_task_def['task_id'] = 8
+        compute_task_def['deadline'] = -11
+        task_to_compute = message.TaskToCompute(
+            timestamp = self.message_timestamp,
+            compute_task_def = self.compute_task_def,
+        )
+
+        serialized_task_to_compute      = dump(task_to_compute,             REQUESTOR_PRIVATE_KEY,   PROVIDER_PUBLIC_KEY)
+        deserialized_task_to_compute    = load(serialized_task_to_compute,  PROVIDER_PRIVATE_KEY,  REQUESTOR_PUBLIC_KEY, check_time = False)
+
+        ack_report_computed_task = message.AckReportComputedTask(
+            timestamp = self.message_timestamp,
```

Comment:
:+1: But personally I think it doesn't make any sense.

Reply:
Why? I'm not strongly opposed to doing it like this. And if there's more code to move to `setUp()` I'm all for it. But in this particular case moving just the timestamp to `setUp()` strikes me as something that only makes it hard to see what the timestamp is.

It would be fine if you moved the whole `AckReportComputed` task to `setUp()`. But why just the timestamp? It's not some generic constant that we can use for all messages. Whether the timestamp is correct or not depends on the other timestaps in the test and they're defined here and not in `setUp()`.